CONTRIBUTING.md

```diff
@@ -0,0 +1,183 @@
+[dokku@josediazgonzalez.com](mailto:dokku@josediazgonzalez.com),
+
+Security reports are greatly appreciated and we will publicly thank you for it.
+
+## Reporting other issues
+
+A great way to contribute to the project is to send a detailed report when you
+encounter an issue. We always appreciate a well-written, thorough bug report,
+and will thank you for it!
+
+Sometimes dokku is missing a feature you need. In some cases, those features can
+be found in pre-existing [plugins](http://progrium.viewdocs.io/dokku/plugins/),
+and we encourage our users to create and contribute such packages. From time to
+time, we will also pull plugins into the dokku core when the task they solve is
+a common one for our users.
+
+Check that [our issue database](https://github.com/progrium/dokku/issues)
+doesn't already include that problem or suggestion before submitting an issue.
+If you find a match, add a quick "+1" or "I have this problem too." Doing this
+helps prioritize the most common problems and requests.
+
+When reporting issues, please include your host OS (Ubuntu 14.04, Fedora 19,
+etc). Please include:
+
+* The output of `uname -a`.
+* The output of `docker version`.
+* The output of `docker -D info`.
+* The output of `dokku version`.
+
+Please also include the steps required to reproduce the problem if possible and
+applicable. This information will help us review and fix your issue faster.
+
+**Issue Report Template**:
+
+```
+Description of problem:
+
+
+Output of the following commands
+
+- `dokku version`:
+- `dokku plugins`:
+- `docker version`:
+- `docker info`:
+- `uname -a`:
+- `cat /home/dokku/<app>/nginx.conf` (if applicable):
+- Output of failing dokku commands with `dokku trace on`:
+
+Environment details (AWS, VirtualBox, physical, etc.):
+
+How was dokku installed?:
+
+How reproducible:
+
+
+Steps to Reproduce:
+1.
+2.
+3.
+
+Actual Results:
+
+
+Expected Results:
+
+
+Additional info:
+```
+
+## Contributing
+
+Before you contribute to the dokku project, there are a few things that you'll
+need to do
+
+* Make sure you have a [GitHub account](https://github.com/signup/free).
+* Submit an [issue](https://github.com/progrium/dokku/issues), assuming one
+  does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub.
+
+### Making Changes
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * Only target an existing branche if you are certain your fix must be on that
+    branch.
+  * To quickly create a topic branch based on master; `git branch
+    master/my_contribution master` then checkout the new branch with `git
+    checkout master/my_contribution`. Better avoid working directly on the
+    `master` branch, to avoid conflicts if you pull in updates from origin.
```

Comment:
maybe reference `git checkout -b` as a single step instead of branch and checkout